src/curl.c

```diff
@@ -164,6 +164,11 @@ int swupd_curl_init(void)
 	int ret;
 	struct stat st;
 
+	if (curl) {
+		error("curl has already been initialized.\n");
+		return -1;
+	}
```

Comment:
what is the purpose of returning a -1 if curl was already initialized? shouldn't it just return with a 0 if it is already initialized and ready to be used? with a -1 there will be no way to distinguish between an already initialized curl and one that failed to initialize (code-wise).

Reply:
Hmmm. That's a good question. I was considering multiple inits as an error, but we support multiple deinits too we can return 0 here. I'll change that.